### PR TITLE
Add readthedocs config

### PR DESCRIPTION
**Pull request by choldgraf**
Explicitly adds a readthedocs configuration file, so that we can pin to Python 3.8 (and because this is a best-practice for RTD)

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+
+python:
+  version: "3.8"
```
Comment (choldgraf) [inline .readthedocs.yml:4]:
```suggestion
  version: "3.8"  # To ensure that we can install the sphinx-2i2c-theme
```

File: .readthedocs.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: 2
+
+python:
+  version: "3.8"  # To ensure that we can install the sphinx-2i2c-theme
+  install:
+    - requirements: requirements.txt
+
```
Comment (consideRatio) [inline .readthedocs.yml:7]:
```suggestion
# Configuration on how ReadTheDocs (RTD) builds our documentation
# ref: https://readthedocs.org/projects/2i2c-team-compass/
# ref: https://docs.readthedocs.io/en/stable/config-file/v2.html
version: 2

build:
  os: ubuntu-20.04
  tools:
    python: "3.8"

sphinx:
  configuration: conf.py

# Explicitly opt out of trying to build additional formats such as PDF and ePub
formats: []

python:
  install:
    - requirements: requirements.txt
```

If you want to ever update to Python 3.9, you must use the non-deprecated way of installing Python, which is like this.

I added some misc stuff I think can be helpful to have declared in a RTD file as well by copy-pasting from other configs.

**Comment by choldgraf** [comment]
hmm - the sphinx-2i2c-theme is now failing to build due to the enviroment missing `node`, but this was working with the old configuration style (https://readthedocs.org/projects/2i2c-team-compass/builds/15282403/)...any idea why that would have changed?

**Comment by consideRatio** [comment]
Hmmm, I figure the build.os -> ubuntu:20.04 makes them use a different base environment?

File: .readthedocs.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Configuration on how ReadTheDocs (RTD) builds our documentation
+# ref: https://readthedocs.org/projects/2i2c-team-compass/
+# ref: https://docs.readthedocs.io/en/stable/config-file/v2.html
+version: 2
+
+build:
+  os: ubuntu-20.04
+  tools:
+    # The sphinx-2i2c-theme requires python >= 3.8
+    python: "3.8"
```
Comment (consideRatio) [inline .readthedocs.yml:10]:
```suggestion
    python: "3.8"
# Set the version of Python and other tools you might need
build:
  os: ubuntu-20.04
  tools:
    python: "3.9"
    nodejs: "16"
```

**Comment by choldgraf** [comment]
well the sphinx-theme-builder should automatically download a node environment to the home directory, which is why I'm confused

File: .readthedocs.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+sphinx:
+  configuration: conf.py
+
+# Explicitly opt out of trying to build additional formats such as PDF and ePub
+formats: []
+
+python:
+  install:
+    - requirements: requirements.txt
```
Comment (consideRatio) [inline .readthedocs.yml:20]:
```suggestion
    - requirements: requirements.txt

```

**Comment by choldgraf** [comment]
yep that seemed to be the issue. I've opened up https://github.com/pradyunsg/sphinx-theme-builder/issues/3 as a result but I think for now we can just define node in the config like you've done here

**Review by consideRatio — APPROVED** [review]

**Comment by choldgraf** [comment]
Merging this since it un-breaks our docs builds!

**Pull request merged**